src/common/interop/two_way_pipe_message_ipc.cpp

```diff
@@ -347,81 +349,49 @@ HANDLE TwoWayPipeMessageIPC::TwoWayPipeMessageIPCImpl::create_medium_integrity_t
 
 void TwoWayPipeMessageIPC::TwoWayPipeMessageIPCImpl::handle_pipe_connection(HANDLE input_pipe_handle)
 {
-    //Adapted from https://docs.microsoft.com/en-us/windows/win32/ipc/multithreaded-pipe-server
-    HANDLE hHeap = GetProcessHeap();
-    uint8_t* pchRequest = (uint8_t*)HeapAlloc(hHeap, 0, BUFSIZE * sizeof(uint8_t));
-
-    DWORD cbBytesRead = 0, cbReplyBytes = 0, cbWritten = 0;
-    BOOL fSuccess = FALSE;
-
-    // Do some extra error checking since the app will keep running even if this thread fails.
-    std::list<std::vector<uint8_t>> message_parts;
-
-    if (input_pipe_handle == NULL)
-    {
-        if (pchRequest != NULL)
-            HeapFree(hHeap, 0, pchRequest);
-        return;
-    }
-
-    if (pchRequest == NULL)
+    if (!input_pipe_handle)
     {
         return;
     }
-
-    // Loop until done reading
+    constexpr DWORD readBlockBytes = BUFSIZE;
+    std::wstring message;
+    size_t iBlock = 0;
+    message.reserve(BUFSIZE);
+    bool ok;
     do
     {
-        // Read client requests from the pipe. This simplistic code only allows messages
-        // up to BUFSIZE characters in length.
-        ZeroMemory(pchRequest, BUFSIZE * sizeof(uint8_t));
-        fSuccess = ReadFile(
-            input_pipe_handle, // handle to pipe
-            pchRequest, // buffer to receive data
-            BUFSIZE * sizeof(uint8_t), // size of buffer
-            &cbBytesRead, // number of bytes read
-            NULL); // not overlapped I/O
-
-        if (!fSuccess && GetLastError() != ERROR_MORE_DATA)
+        constexpr size_t charsPerBlock = readBlockBytes / sizeof(message[0]);
+        message.resize(message.size() + charsPerBlock);
+        DWORD bytesRead = 0;
+        ok = ReadFile(
+            input_pipe_handle,
+            message.data() + iBlock * charsPerBlock,
+            readBlockBytes,
+            &bytesRead,
```

Comment:
SG: Leave a comment "bytesRead is either equal to readBlockBytes or it was the last read from the pipe". Without the comment line `message.data() + iBlock * charsPerBlock` can be confusing.